src/arch/xtensa/boot_loader.c

```diff
@@ -219,14 +219,17 @@ static uint32_t hp_sram_init(void)
 
 #endif
 
-#if defined(CONFIG_ICELAKE)
+#if defined(CONFIG_ICELAKE) || defined(CONFIG_APOLLOLAKE)
```

Comment:
In line 237 definition of this shim register LSPGISTS is missing for APOLLOLAKE.